src/requests/consumption.js

```diff
@@ -0,0 +1,1059 @@
+						if (!metaInformation.recommendedFor.includes(role.label))
+							metaInformation.recommendedFor.push(role.label)
+					})
+				} else {
+					scope.roles = []
+					metaInformation.recommendedFor = []
+				}
+				targeting[targeting_entity].forEach((target) => {
+					if (scope[targeting_entity] == undefined) scope[targeting_entity] = []
+					if (metaInformation[targeting_entity] == undefined) metaInformation[targeting_entity] = []
+					metaInformation[targeting_entity].push(target.name)
+					scope[targeting_entity].push(target._id)
+				})
+			})
+		}
+
+		let template = {
+			scope,
+			metaInformation,
+			resourceType: [common.ROLLOUT_TYPE_PROGRAM],
+			language,
+			keywords,
+			concepts: templateData?.concepts ? templateData?.concepts : [],
+			components: [],
+			resourceDetails,
+			isAPrivateProgram: false,
+			isDeleted: false,
+			requestForPIIConsent: templateData?.requestForPIIConsent ? true : false,
+			rootOrganisations: [
+				templateData?.rootOrganisations ? templateData?.rootOrganisations : templateData?.organization?.id,
+			],
+			createdFor: [templateData?.createdFor ? templateData?.createdFor : templateData?.organization?.id],
+			deleted: false,
+			status: common.STATUS_ACTIVE.toLowerCase(),
+			owner: templateData?.created_by,
+			createdBy: templateData?.created_by,
+			updatedBy: templateData?.created_by,
+			externalId: generateExternalId(templateData?.title),
+			name: templateData?.title.trim(),
+			description: templateData?.description ? templateData?.description : '',
+			updatedAt: new Date(),
+			createdAt: new Date(),
+			endDate: new Date(templateData?.end_date),
+			startDate: new Date(templateData?.start_date),
+		}
+
+		return { success: true, template }
+	} catch (error) {
+		console.error('Error in formatTemplate:', error.message)
+		return { success: false, error: error.message }
+	}
+}
+
+// function to replace special charecters
+const escapeXml = (unsafe) => {
+	return unsafe
+		.replace(/&/g, '&amp;')
+		.replace(/</g, '&lt;')
+		.replace(/>/g, '&gt;')
+		.replace(/"/g, '&quot;')
+		.replace(/'/g, '&apos;')
+}
+
+/**
+ * create svg template by editing base template.
+ * @method
+ * @name createSvg
+ * @param {Object} certificateData - Certificate data for upload
+ */
+
+async function createSvg(certificateData) {
+	return new Promise(async (resolve, reject) => {
+		try {
+			// fetch base template from cloud
+			let baseTemplate = await getBaseTemplate(certificateData?.base_template_url)
+			if (!baseTemplate.success) {
+				throw new Error('Base template download failed.')
+			}
+
+			// Load SVG template using Cheerio with XML mode
+			const $ = cheerio.load(baseTemplate.result, { xmlMode: true })
+
+			// set issuer name
+			const issuerNameTag = 'stateTitle'
+			const issuerNameElement = $(`#${issuerNameTag}`)
+			issuerNameElement.text(escapeXml(certificateData.issuer))
+
+			// update signature
+			for (let index = 1; index <= certificateData.signature.no_of_signature; index++) {
```

Comment:
@adithyadinesh0412  please change the logic for this 
instead of using index, tomorrow classname can be anything so from teh certificate signature get all keys and replace except no_of_signature

sample data 
"certificate": {
            "code": "one_logo_one_sign",
            "signature": {
                "signatureTitleName1": "Smti R S Manners MCS",
                "signatureTitleName2": "",
                "signatureImg2": "",
                "signatureTitleDesignation2": "",
                "signatureImg1": "https://mentoring-prod-storage.s3.ap-south-1.amazonaws.com/certificate/314/certificate/cd9b539b-88ad-4929-a2ce-db6bad2676d0/image-120x50%20%282%29.png",
                "signatureTitleDesignation1": "Director Directorate of Educational Research and Training",
                "no_of_signature": 1
            },
            "base_template_id": 1,
            "criteria": {
                "validationText": "Complete validation message",
                "expression": "C1&&C3",
                "conditions": {
                    "C3": {
                        "validationText": "Evidence task level validation",
                        "expression": "70aa6379-df7f-4033-a653-d58e45adff24&&c43eab58-cb4d-4a6d-a6ee-ab63952e4e43",
                        "conditions": {
                            "c43eab58-cb4d-4a6d-a6ee-ab63952e4e43": {
                                "filter": {
                                    "value": "all",
                                    "key": "type"
                                },
                                "scope": "task",
                                "function": "count",
                                "value": "6",
                                "key": "attachments",
                                "operator": ">=",
                                "taskDetails": [
                                    "c43eab58-cb4d-4a6d-a6ee-ab63952e4e43"
                                ]
                            },
                            "70aa6379-df7f-4033-a653-d58e45adff24": {
                                "filter": {
                                    "value": "all",
                                    "key": "type"
                                },
                                "scope": "task",
                                "function": "count",
                                "value": "8",
                                "key": "attachments",
                                "operator": ">=",
                                "taskDetails": [
                                    "70aa6379-df7f-4033-a653-d58e45adff24"
                                ]
                            }
                        }
                    },
                    "C1": {
                        "validationText": "Project Should be submitted.",
                        "expression": "C1",
                        "conditions": {
                            "C1": {
                                "scope": "project",
                                "value": "submitted",
                                "key": "status",
                                "operator": "=="
                            }
                        }
                    },
                    "C2": {
                        "validationText": "Evidence project level validation",
                        "expression": "C1",
                        "conditions": {
                            "C1": {
                                "filter": {
                                    "value": "all",
                                    "key": "type"
                                },
                                "scope": "project",
                                "function": "count",
                                "value": "1",
                                "key": "attachments",
                                "operator": ">="
                            }
                        }
                    }
                }
            },
            "name": "One Logo One Signature",
            "base_template_url": "https://mentoring-prod-storage.s3.ap-south-1.amazonaws.com/certfile_BASE_TEMPLATE/system/cert/cf317c7c-4496-4fe1-8648-8cf751ec78c6/one_logo_one_sign.svg",
            "logos": {
                "stateLogo1": "https://mentoring-prod-storage.s3.ap-south-1.amazonaws.com/certificate/314/certificate/f9d18893-461b-452e-ad65-633b5d0015d6/image-80x80.png",
                "stateLogo2": "",
                "no_of_logos": 1
            },
            "issuer": "Directorate of Educational Research and Training Government of Meghalaya"
        },